test/e2e_node/podresources_test.go

```diff
@@ -169,7 +174,27 @@ func findContainerDeviceByName(devs []*kubeletpodresourcesv1.ContainerDevices, r
 	return nil
 }
 
-func matchPodDescWithResources(expected []podDesc, found podResMap) error {
+type deviceCheckFunc func(expect podDesc, found podResMap) error
+
+func checkForTopology(expect podDesc, found podResMap) error {
```

Comment:
I'd just inline this check in expectPodResources.
As it is now, if the kubevirt resource name is not found, we silently skip and succeed. Then it makes sense to just always run this check, avoiding the extra argument.

Once the inlining is done, we should add another check: if the resource is not found *but it was expected*, we should fail and error out.

To learn if the resource was expected, we need to make logic, and possibly add a field, to `podDesc`.

Reply:
I'd inline the check around here:
```
			dev := findContainerDeviceByName(cntInfo.GetDevices(), podReq.resourceName)
			if dev == nil {
			if dev == nil {
				return fmt.Errorf("pod %q container %q expected data for resource %q not found", podReq.podName, podReq.cntName, podReq.resourceName)
				return fmt.Errorf("pod %q container %q expected data for resource %q not found", podReq.podName, podReq.cntName, podReq.resourceName)
			}
			}
			if len(dev.DeviceIds) != podReq.resourceAmount {
			if len(dev.DeviceIds) != podReq.resourceAmount {
				return fmt.Errorf("pod %q container %q resource %q expected %d items got %v", podReq.podName, podReq.cntName, podReq.resourceName, podReq.resourceAmount, dev.DeviceIds)
				return fmt.Errorf("pod %q container %q resource %q expected %d items got %v", podReq.podName, podReq.cntName, podReq.resourceName, podReq.resourceAmount, dev.DeviceIds)
			}
			}
```
at the bottom of this `if`.